okhttp/src/main/java/okhttp3/ConnectionPool.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.connection.RealConnectionPool
+import java.util.concurrent.TimeUnit
+
+/**
+ * Manages reuse of HTTP and HTTP/2 connections for reduced network latency. HTTP requests that
+ * share the same [Address] may share a [Connection]. This class implements the policy
+ * of which connections to keep open for future use.
+ *
+ * @constructor Create a new connection pool with tuning parameters appropriate for a single-user
+ * application. The tuning parameters in this pool are subject to change in future OkHttp releases.
+ * Currently this pool holds up to 5 idle connections which will be evicted after 5 minutes of
+ * inactivity.
+ */
+class ConnectionPool @JvmOverloads constructor(
```

Comment:
Java version only had two constructors, the no-arg and the all-arg. `@JvmOverloads` will instead produce 4 constructors. Are we fine with it, or should I change it to two constructors declared explicitly?

Reply:
Two explicit constructors. The variations produced by `JvmOverloads` don't make sense. You always must pass a `TimeUnit` when supplying a time value

Reply:
my 2c: the @JvmOverloads approach is better as its source and binary compatible, but sets a nicer precedent.

Reply:
hmmm....., weird to specify keepAliveDuration but not timeUnit.  So worth discussion.

Reply:
@JakeWharton right, I didn't notice that. Will revert to two constructors and can follow up with a more idiomatic version if needed.